processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperLogLogCollector.java

```diff
@@ -640,7 +640,7 @@ private short addNibbleRegister(short bucket, byte positionOf1)
    * @param offsetDiff The difference in offset between the byteToAdd and the current HyperLogLogCollector
    * @param byteToAdd  The byte to merge into the current HyperLogLogCollector
    */
-  private static int mergeAndStoreByteRegister(
+  private static short mergeAndStoreByteRegister(
```

Comment:
Its callers in this method store the results as `short` this make sure its ~~turtles~~ `short` all the way down